### DISCUSSION TITLE
Unused keys in container shema

### DISCUSSION THREAD

**Discussion by docjyJ** [nextcloud/all-in-one#5574] [Ideas] [answered]
### Steps to reproduce
1. In file, https://github.com/nextcloud/all-in-one/blob/main/php/containers-schema.json: `ctrl+f` > `cap_drop`, `profiles`, `user`, `networks` or `expose`
2. In file, https://github.com/nextcloud/all-in-one/blob/main/php/src/ContainerDefinitionFetcher.php: `ctrl+f` > `cap_drop`, `profiles`, `user`, `networks` or `expose`
3. They are defined but never used...

https://github.com/nextcloud/all-in-one/blob/267aebf941883bc21437e51974d14079aada1bc8/php/containers.json#L73-L75

#### Other valuable info 

*(I'm rewriting mastercontainer in rust (to learn rust) and I found this...)*

**Comment by docjyJ** [comment]
Maybe this behavior is intended for yaml, but it's confusing.

**Comment by szaimen** [comment]
Yes, it is intended because we generate https://github.com/nextcloud/all-in-one/blob/main/manual-install/latest.yml and https://github.com/nextcloud/all-in-one/tree/main/nextcloud-aio-helm-chart out of it. Do you have a better idea how to do that?

**Comment by docjyJ** [comment]
Why have a cap_drop in YAML, but not with the mastercontainer ?

**Comment by szaimen** [comment]
Because cap_drop is by default applied for all containers: https://github.com/nextcloud/all-in-one/blob/267aebf941883bc21437e51974d14079aada1bc8/php/src/Docker/DockerActionManager.php#L519

**Comment by szaimen** [comment]
Or WDYM?

**Comment by szaimen** [comment]
Linking to https://github.com/nextcloud/all-in-one/pull/3377 which was introduced first. Later https://github.com/nextcloud/all-in-one/pull/4116 was created.

**Comment by szaimen** [comment, marked as answer]
`cap_drop`, `profiles`, `user`, `networks` or `expose`

Actually, all of them are only there for the docker-compose yaml and the helm chart because they are generated out of the containers.json. In the mastercontainer itself is a separate logic to handle these, thus they are not actually needed for the mastercontainer.

How should we proceed with this issue?

**Reply by szaimen** [reply]
Just FYI: there will be an additional key (healthcheck) soon: https://github.com/nextcloud/all-in-one/pull/5689/files

**Reply by docjyJ** [reply]
good!

**Reply by szaimen** [reply]
Just FYI: the networks key is getting removed via https://github.com/nextcloud/all-in-one/pull/5862

**Comment by docjyJ** [comment]
Move this to discussions as ideas?

**Comment by szaimen** [comment]
> Move this to discussions as ideas?

Sounds good 👍

**Comment by docjyJ** [comment]
I found this which is not used:
https://github.com/nextcloud/all-in-one/blob/9341392c00845beba521813323022108f55c28ee/php/src/Auth/AuthManager.php#L30

**Reply by docjyJ** [reply]
Search:
https://github.com/search?q=repo%3Anextcloud%2Fall-in-one%20date_time&type=code

**Reply by szaimen** [reply]
It is used here: https://github.com/nextcloud/all-in-one/blob/main/Containers/mastercontainer/session-deduplicator.sh

**Comment by docjyJ** [comment]
This files is nevr read: https://github.com/nextcloud/all-in-one/blob/9341392c00845beba521813323022108f55c28ee/php/src/Auth/AuthManager.php#L37

**Reply by docjyJ** [reply]
Search: https://github.com/search?q=repo%3Anextcloud%2Fall-in-one+GetSessionDateFile&type=code

**Reply by docjyJ** [reply]
So I'm stuck here:
https://github.com/docjyJ/all-in-one-rust/blob/14b800ee7f62b7bdbe9829c3104d525f8eaf959b/src/auth/controller.rs#L30-L45
In the end, I have the impression that only the boolean variable is used...

**Reply by szaimen** [reply]
It is used here: https://github.com/nextcloud/all-in-one/blob/main/Containers/mastercontainer/session-deduplicator.sh
